File: hello-world/src/main/java/com/example/helloworld/controller/HelloController.java
package com.example.helloworld;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

    @GetMapping("/hello")
    public String helloWorld(@RequestParam(defaultValue = "Guest") String name) {
        return "Hello, " + name + "!";
    }
}

File: kotlin-hello/src/main/java/com/example/kotlinhello/HelloController.kt
package com.example.kotlinhello

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class HelloController {

    @GetMapping("/hello")
    fun sayHello(): String {
        return "Hello, Kotlin with Spring Boot 3!"
    }
}